src/MetaMorpho.sol

```diff
@@ -328,9 +328,12 @@ contract MetaMorpho is ERC4626, ERC20Permit, Ownable2Step, Multicall, IMetaMorph
             if (!seen[i]) {
                 Id id = withdrawQueue[i];
 
-                if (MORPHO.supplyShares(id, address(this)) != 0 || config[id].cap != 0) {
-                    revert ErrorsLib.InvalidMarketRemoval(id);
-                }
+                if (config[id].cap != 0) revert ErrorsLib.InvalidMarketRemovalNonZeroCap(id);
+
+                try this.expectedSupplyAssets(_marketParams(id)) returns (uint256 supplyAssets) {
+                    // Assumes that a market with zero supply assets (but possibly non-zero supply shares) is dust.
+                    if (supplyAssets != 0) revert ErrorsLib.InvalidMarketRemovalNonZeroSupply(id);
+                } catch {}
```

Comment:
```suggestion
                if (isAllocator[_msgSender()] && MORPHO.supplyShares(id) != 0) revert ErrorsLib.InvalidMarketRemovalNonZeroSupply(id);
```

Reply:
isn't curator that should do it instead?

Reply:
Oh right, I had the opposite in my head. So indeed if it's the allocator that triggers the function we perform the check.

Don't we want to do the try/catch for curator/owner though? Else it's very easy to remove a market

Reply:
As stated by @QGarchery, the try/catch is vulnerable to OOG, so it wouldn't be a valid additional safety measure

Reply:
> How would an amount of total gas exist such that the call to Morpho reverts, considering the EVM forwards 63/64 of remaining gas to Morpho? @QGarchery

I'm not sure that it is feasible, but I think that the idea is to not have enough gas with 63/64 of the remaining gas to complete the morpho call, but have enough gas with 1/64 to complete the loop (which could potentially happen if you put the faulty market at the end)